src/subcommands/add.go

```diff
@@ -1,184 +1,228 @@
 }
 
-func commitAdd(ale AddedListEntry, fp *os.File) error {
-	// Seek to the end of the file
-	_, err := fp.Seek(0, io.SeekEnd)
-	if err != nil {
-		return fmt.Errorf("failed to seek to end of file: %w", err)
+
+// commitAdd appends the provided IndexListEntry to the index file specified by indexFilePath.
+// It reads the existing index file data, unmarshals it, appends the provided entry, marshals the updated index data,
+// and writes it back to the index file. It returns an error if any operation fails.
+//
+// Parameters:
+//   ile *pb.IndexListEntry: The IndexListEntry to be appended to the index entries.
+//   indexFilePath string: The file path of the index file.
+//
+// Return:
+//   error: An error if any operation fails.
+func commitAdd(ile *pb.IndexListEntry, indexFilePath string) error {
+
+	var index pb.Index
+	// Read the existing index file data
+	data, err := os.ReadFile(indexFilePath)
+
+	if err == nil {
+		// Unmarshal the existing index data
+		if err := proto.Unmarshal(data, &index); err != nil {
+			return fmt.Errorf("failed to unmarshal existing index: %w", err)
+		}
+	} else if !os.IsNotExist(err) {
+		// Return an error if failed to read the index file and the error is not due to the file not existing
+		return fmt.Errorf("failed to read index file: %w", err)
 	}
 
-	// Create an encoder that will append to the file
-	encoder := gob.NewEncoder(fp)
+	// Append the provided IndexListEntry to the index entries
+	index.Entries = append(index.Entries, ile)
 
-	// Encode and append the new entry
-	err = encoder.Encode(&ale)
+	// Marshal the updated index data
+	updatedData, err := proto.Marshal(&index)
 	if err != nil {
-		return fmt.Errorf("failed to encode and append new entry: %w", err)
+		return fmt.Errorf("failed to marshal updated index: %w", err)
+	}
+	// Write the updated index data to the index file
+	if err := os.WriteFile(indexFilePath, updatedData, 0644); err != nil {
+		return fmt.Errorf("failed to write updated index to file: %w", err)
 	}
 
 	return nil
 }
 
-func copyFile(src string, dest string, ale AddedListEntry) error {
-	// Open the source file
+
+// copyFile copies the content of the source file to the destination file and updates the given IndexListEntry with the content and last refreshed time.
+// Parameters:
+//   src: the path of the source file to be copied
+//   dest: the path of the destination file where the content will be copied to
+//   ile: a pointer to an IndexListEntry to be written to the destination file
+// Return:
+//   error: an error if any operation fails during the file copying process
+func copyFile(src string, dest string, ile *pb.IndexListEntry) error {
+	// Open the source file for reading
 	srcFile, err := os.Open(src)
 	if err != nil {
 		return fmt.Errorf("failed to open source file: %w", err)
 	}
-	defer srcFile.Close()
+	defer srcFile.Close() // Close the source file when the function returns
 
-	// Read the entire file content
+	// Read all content from the source file
```

Comment:
Remove this comment (in general let's not leave comments of this level of obviousness)

Reply:
I realize some of these comments already exist from when I added them, but just might as well clean them up now